apps/client-test/Main.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main where
+
+import Control.Concurrent
+import Control.Exception
+import Control.Monad
+import GHC.IO.Exception
+import Network.Socket
+import System.IO
+import System.IO.Error
+import Util
+
+main :: IO ()
+main = runTCPClient "localhost" "5000" ping
+
+ping :: Handle -> IO ()
+ping h = do
+  putStrLn "connected to server"
+  forever $ do
+    putLn h "ping"
+    threadDelay 1000000
+    line <- getLn h
+    print line
+
+runTCPClient :: HostName -> ServiceName -> (Handle -> IO a) -> IO a
```

Comment:
Is there a way to make `runTCPClient` from Transport more generic? If not maybe move this one to Transport as well and rename the one there to `runTCPClientUnliftIO` to increase chances to avoid code divergence?

Also `startTCPClient` is very similar apart from throwing exception, but it doesn't have latest changes from Transport.